diff --git a/lib/moodlelib.php b/lib/moodlelib.php
index 51f2e839..9b14219 100644
--- a/lib/moodlelib.php
+++ b/lib/moodlelib.php
@@ -5280,6 +5280,13 @@ function reset_course_userdata($data) {
         comment::reset_course_page_comments($context);
     }

+    // SUP-5040.  Probably the core course_reset event should be used when it becomes available (2.7?)
+    //            and this event can be removed:
+    events_trigger('local_course_reset', (object) array(
+        'courseid'  => $data->courseid,
+        'startdate' => empty($data->reset_start_date) ? null : $data->reset_start_date,
+    ));
+
     return $status;
 }

